Frontend/src/views/ResetPassword.vue

```diff
@@ -0,0 +1,173 @@
+      newPassword: null,
+      networkErrorMessage: null,
+      haveNetworkError: false,
+      wrongAnswerCounter : 0,
+      successAlert: null,
+      failAlert: null
+    }
+  },
+  created () {
+    axios({
+      method: 'GET',
+      url: `${apiURL}/reset/` + this.resetToken,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Credentials': true
+      }
+    })
+      .then(response => (this.securityQuestions = response.data),
+        this.message = 'Enter your answers for the security questions, fields are case sensitive')
+      .catch(e => { this.message = e.response.data }, this.haveNetworkError = true)
+    if(this.message === "Reset link is no longer valid"){
+      this.$router.push("SendResetLink")
+    }
+  },
+  methods: {
+    redirectToLogin: function () {
+      this.$router.push( "/login" )
+    }, 
+    submitAnswers: function () {
+      if(this.wrongAnswerCounter === 3){
+        this.errorMessage = "3 attempts have been made, reset link is no longer valid"
+        this.$router.push("SendResetLink")
+      }
+      if (!this.securityAnswer1 || !this.securityAnswer2 || !this.securityAnswer3){
+        alert("Security answers cannot be empty")
+      } else {
+        axios({
+        method: 'POST',
+        url: `${apiURL}/reset/` + this.resetToken + '/checkanswers',
+        data: { 
+          securityA1: this.$data.securityAnswer1,
+          securityA2: this.$data.securityAnswer2,
+          securityA3: this.$data.securityAnswer3},
+        headers: {
+          'Access-Control-Allow-Origin': '*',
+          'Access-Control-Allow-Credentials': true
+        }
+      })
+        .then(response => (this.showPasswordResetField = response.data))
+        .catch(e => { this.message = e.response.data }, this.haveNetworkError = true)
+      if (this.showPasswordResetField === false) {
+        this.errorMessage = "Answers are incorrect"
+        this.wrongAnswerCounter = this.wrongAnswerCounter + 1
+      }
+      }
+    },
+    submitNewPassword: function () {
+      if(this.newPassword === null){
+        alert("Password cannot be empty")
+      } else if (this.newPassword.length < 12){
+        alert("Password must be at least 12 characters")
+      } else if (this.newPassword.length > 2000) {
+        alert("Password must be less than 2000 characters")
+      } else {
+        axios({
+        method: 'POST',
+        url: `${apiURL}/reset/` + this.resetToken + '/resetpassword',
+        data: {newPassword: this.$data.newPassword},
+        headers: {
+          'Access-Control-Allow-Origin': '*',
+          'Access-Control-Allow-Credentials': true
+        }
+      })
+        .then(response => (this.message = response.data), this.successAlert = true, 
+        setTimeout(() => this.redirectToLogin(), 3000))
+        .catch(e => { this.message = e.response.data }, this.haveNetworkError = true, this.fail = true)
+      }
+    }
+  }
+}
+</script>
+
+<style lang="css">
```

Comment:
this css is overwriting other components, makes it look ugly. Make this UI match login and register, etc